client/src/error.rs

```diff
@@ -27,6 +27,7 @@ pub enum Error {
     Io(io::Error),
     InvalidAmount(bitcoin::util::amount::ParseAmountError),
     InvalidCookieFile,
+    SubmitBlock(String),
```

Comment:
Hmm, it's a bit silly to have a error variant that only a single call can return. I think there's other calls that have a "plain string error type" return type, so we might generalize that with an `ErrorString(String)` variant or something?

Reply:
Of course, I'd be happy to change that. However, I'm not sure if this somehow conflicts with the expectation a user aware of the actual JSON API would have. The call `submitblock` seems to be the odd one out there, as it returns `Null` in case of success and a `String` in case of an error. While I agree that having a dedicated `SubmitBlock` error type is not very efficient, it at least accounts for the fact that this call somehow behaves different than the other ones?

Reply:
> a bit silly to have a error variant that only a single call can return

How about making this a separate single-variant error type instead of a variant of the main `Error` type, something like `struct SubmitBlockError(String)`?

I would personally prefer a more explicit error type over a generic `ErrorString`.

Reply:
While this still feels a bit clunky, I now pushed an update with a dedicated `SubmitBlockError(String)` error type.

Reply:
For the sake of consistency (and so the user can use these calls in a similar fashion), I now also introduced a dedicated `SubmitBlockResult` result type. Not sure if this is a good idea though, feels bloated for such a small application.

Reply:
I don't think that type is needed, especially since it's not much shorter and only used in ~4 places. Also: if you still want to do this you don't need to make it generic since `T` will always be `()` (which also makes the build fail rn because you didn't specify `T` for the return types).

Reply:
Whoops, sorry for that. Apart from that you are also probably right. I reverted the last commit.